src/nvim/os/time.c

```diff
@@ -46,26 +47,45 @@ void os_delay(uint64_t milliseconds, bool ignoreinput)
     }
     LOOP_PROCESS_EVENTS_UNTIL(&main_loop, NULL, (int)milliseconds, got_int);
   } else {
-    os_microdelay(milliseconds * 1000);
+    os_microdelay(milliseconds * 1000u, ignoreinput);
```

Comment:
In the old code this branch did not allow any interruptions.  Now any keypress can interrupt the wait.